src/mcp_agent/utils/resource_utils.py

```
@@ -0,0 +1,222 @@
+    This creates an EmbeddedResource that references another resource URI.
+    When the client receives this, it will make a separate request to fetch
+    the resource content using the provided URI.
+
+    Args:
+        uri: URI for the resource
+        mime_type: MIME type of the resource
+
+    Returns:
+        An EmbeddedResource object
+    """
+    from mcp.types import EmbeddedResource, TextResourceContents
```

Pylint (3.3.7)

[warning] Redefining name 'EmbeddedResource' from outer scope (line 5)

(W0621)